client/state/plugins/wporg/reducer.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import React from 'react/addons';
+
+function updatePluginState( state = {}, pluginSlug, attributes ) {
+	return Object.assign( {},
+		state,
+		{ [ pluginSlug ]: Object.assign( {}, state[ pluginSlug ], attributes ) }
+	);
+}
```

Comment:
It's really mostly for convenience and readability. I get the concern about bringing in an external helper that arguably adds some level of complexity. What prompted my suggestion was the case of `'RECEIVE_WPORG_PLUGIN_DATA'`, which has an `assign` within an `assign` — seemingly, a good candidate for `update`. Totally up to you. :)